pyfr/plugins/tavg.py

```diff
@@ -358,3 +363,248 @@ def __call__(self, intg):
+                    self.std_avg = np.zeros(len(anames), dtype=self.dtype)
+                    self.fstd_max = np.zeros(len(fnames), dtype=self.dtype)
+                    self.fstd_avg = np.zeros(len(fnames), dtype=self.dtype)
+
+                # Build fun avg expressions
+                c = cfg.items_as('constants', float)
+                self.fexprs = fexprs = []
+                for k in fnames:
+                    fexprs.append(cfg.getexpr(cfgsect, f'fun-avg-{k}', subs=c))
+            else:
+                self.mapping.append((amap, smap))
+
+            # Check for compatibility of files
+            if self.uuid != file['mesh_uuid']:
+                raise RuntimeError('Average files computed on different meshes')
+            if self.region != cfg.get(cfgsect, 'region'):
+                raise RuntimeError('Average files computed on different regions')
+
+            for k in cfg.items(cfgsect, prefix='avg-'):
+                if cfg.get(cfgsect, k) != self.cfg.get(self.cfgsect, k):
+                    raise RuntimeError('Different average field definitions')
+
+    def _merge_data(self, outf):
+        file0, stats, cfg, dt = self.files[0]
+        amap0, smap0 = self.mapping[0]
+
+        for key in self.dkeys:
+            # Initialise accumulators
+            data = file0[key]
+            avg_acc = np.zeros_like(data[:, amap0], dtype=np.float64())
+            var_acc = np.zeros_like(data[:, smap0], dtype=np.float64())
+
+            # Perform the rest of the accumulations
+            for files, mapping in zip(self.files, self.mapping):
+                file, stats, cfg, dt = files
+                amap, smap = mapping
+                data = file[key]
+
+                avg_acc += dt*data[:, amap]
+                if self.std_all:
+                    var_acc += dt*(data[:, smap]**2 + data[:, amap]**2)
+
+            # Shift variance and get standard deviation
+            if self.std_all:
+                var_acc = np.sqrt(var_acc - avg_acc**2)
+                np.maximum(self.std_max, np.amax(var_acc, axis=(0, 2)),
+                           out=self.std_max)
+                self.std_avg += np.mean(var_acc, axis=(0, 2))
+
+            # Evaluate function expression and write out
+            if self.fexprs and self.std_all:
+                fun_avg, fun_std = self._eval_fun_var(var_acc, avg_acc)
+                np.maximum(self.fstd_max, np.amax(fun_std, axis=(0, 2)),
+                           out=self.fstd_max)
+                self.fstd_avg += np.mean(fun_std, axis=(0, 2))
+
+                outf[key] = np.hstack((avg_acc, fun_avg, var_acc, fun_std),
+                                       dtype=self.dtype)
+            elif self.fexprs:
+                fun_avg = self._eval_fun_exprs(avg_acc)
+                outf[key] = np.hstack((avg_acc, fun_avg, var_acc),
+                                      dtype=self.dtype)
+            else:
+                outf[key] = np.hstack((avg_acc, var_acc), dtype=self.dtype)
```

Comment:
What happens when `std_all` is False?

Reply:
then var_acc is empty and it all works out fine.